src/hl/group.rs

```diff
@@ -6,8 +6,8 @@ use hdf5_sys::{
     h5d::H5Dopen2,
     h5g::{H5G_info_t, H5Gcreate2, H5Gget_info, H5Gopen2},
     h5l::{
-        H5L_info_t, H5L_iterate_t, H5Lcreate_hard, H5Lcreate_soft, H5Ldelete, H5Lexists,
-        H5Literate, H5Lmove, H5L_SAME_LOC,
+        H5L_info1_t as H5L_info_t, H5L_iterate1_t as H5L_iterate_t, H5Lcreate_hard, H5Lcreate_soft,
```

Comment:
Maybe we should use the v2 variants when these are available?

Reply:
I think we have to, but then you'll end up with lots of `cfg_attr` boilerplate in hdf5 crate code. I was even thinking of some helper macros here or aliases, will it help? Hm.

Reply:
We will have to do this on a function-to-function basis I think. Since the topcrate is higher level than `hdf5-sys`, we can probably hide the different versions inside each function.